validator/src/types.rs

```diff
@@ -20,6 +20,13 @@ impl ValidationError {
     pub fn add_param<T: Serialize>(&mut self, name: Cow<'static, str>, val: &T) {
         self.params.insert(name, to_value(val).unwrap());
     }
+
+    /// Adds a custom message to a `ValidationError` that will be used when displaying the
+    /// `ValidationError`, instead of an auto-generated description.
+    pub fn with_message(mut self, message: std::borrow::Cow<'static, str>) -> ValidationError {
```

Comment:
You don't need the full path to Cow, it's already imported

Reply:
Good point. Full path removed.

The project I'm working for during my day job, we've decided to never import anything on module-level and always use the fully qualified paths. Imports within scopes is fine, if it helps with legibility. Comes with its own pros and cons. Now it's just a habit to always do that.